board_client/animations/checkerboard_pulse.py

```
@@ -1,34 +1,142 @@
+                rotated_x = (dx * cos_angle - dy * sin_angle) // SCALE + current_params["scroll_x_scaled"]
+                rotated_y = (dx * sin_angle + dy * cos_angle) // SCALE + current_params["scroll_y_scaled"]
+
+                checker_x = rotated_x // size_scaled
+                checker_y = rotated_y // size_scaled
+                is_checker = (checker_x + checker_y) % 2 == 0
+
+                hsv1 = current_params["hsv1"]
+                hsv2 = current_params["hsv2"]
+
+                if in_transition and next_params:
+                    progress = min(1.0, (current_time_s - transition_start_time) / FADE_DURATION_S)
```

Pylint (3.3.7)

[error] Possibly using variable 'transition_start_time' before assignment

(E0606)